### ISSUE TITLE
Multipass: Add a section for Linux under "How to troubleshoot networking"

### ISSUE THREAD

**Issue by giuliazanchi** [canonical/open-documentation-academy#113]
## Context

[Multipass docs](https://multipass.run/docs) are in reasonably good shape, but there are (as always) little things that could be improved to offer a better user experience. 

Multipass docs are maintained in and published from the Ubuntu Discourse forum. To suggest changes in a Discourse-hosted docs page, scroll all the way down to the bottom of the page and open the page to which the "Help improve this document in the forum" link points. On the Discourse page, scroll to the bottom of the article (if there are comments, then the bottom of the article is just where the comments start, i.e. not at the very bottom of the page) and click "Edit". (You need an account on Ubuntu Discourse.) See also: [Contribute to Multipass docs](https://multipass.run/docs/contribute-to-multipass-docs).

## Task

The [How to troubleshoot networking](https://multipass.run/docs/troubleshoot-networking) guide is missing a Linux section. To create it, you could leverage your own experience using Multipass on Linux, scrape recent bug reports on the [Multipass GitHub](https://github.com/canonical/multipass/issues) for user-reported troubles and solutions, or include potential issues suggested by our dev team such as lingering `dnsmasq` & subnet clashes. 

This task requires some level of familiarity with networking concepts and common issues, as well as experience using Multipass on Linux.

## Useful links

Before you get started, make sure to review the main concepts of the [Diátaxis](https://diataxis.fr/) documentation framework and also check the [Canonical Documentation Style Guide](https://docs.ubuntu.com/styleguide/en/).

If you are new to the Canonical Open Documentation Academy, you may also want to review our [Getting Started](https://discourse.ubuntu.com/t/getting-started/42769) topic on the Ubuntu Discourse forum.

**Comment by vevayos** [comment]
I've been an off an on user of Ubuntu Desktop, and while I was researching Canonical as a company over the last few weeks, I discovered Multipass.  What a wonderful tool!  I'd like to try my hand at working on this issue.  I'm new to Multipass and Diátaxis, but for starters I'll take it to mean that I'll give a first try at putting something out there, and improving it as I learn more.

While I was setting up a Frankenstein computer with Ubuntu 22.04.4, I ran into a snag installing texlive-full.  The installation hung.  I was worried there was something wrong with my new hardware.  I finally found the bug on Launchpad, and I'll also be writing documentation on Ask Ubuntu to address reports about the bug, suggesting to use the texlive-full package in a Multipass instance.

**Comment by giuliazanchi** [comment]
Thank you, @vevayos ! That sounds great. Let me know if you need help with this, I'll be happy to review issues with you or get additional details from our engineers if needed. I'll assign this to you then :D

**Comment by vevayos** [comment]
Thank you, @giuliazanchi!  I may need your help as this is also my first issue.  I forked the repository just now.

**Comment by giuliazanchi** [comment]
Of course, @vevayos ! This particular issue requires understanding of networking issues and troubleshooting practices and workarounds that you may find on the Multipass GitHub, so if it feels a bit too daunting as a first task do not hesitate to reach out. It doesn't have to be perfect - let's aim at improving what we have, even with small updates, and build from there. Thanks!

**Comment by vevayos** [comment]
@giuliazanchi, I'm still working on the issue.  I'll need another week to stop fooling around.

I was researching how the network behavior might be expected to differ depending on which hypervisor, `local.driver`, is used.

The Multipass product page, [https://multipass.run/](https://multipass.run/), says that LXD is used on Linux, but this page, [https://multipass.run/docs/set-up-the-driver](https://multipass.run/docs/set-up-the-driver), says it depends on the capabilities of the architecture, and also the driver can be changed. 

My hardware supports Intel VT and VT-d, and in my Multipass instance,
```
$ systemd-detect-virt
kvm
```
Perhaps `systemd-detect-virt` needs to be revised to return, `lxd`?

Also, my host shows,
```
$ multipass get local.driver
qemu
```
Probably it is only checking if a module is loaded.  But shouldn't this really return, `libvirt`, because in my Multipass instance, it seems to be part of my kernel,
```
$ grep -i virtio /boot/config-6.8.0-41-generic
CONFIG_BLK_MQ_VIRTIO=y
CONFIG_VIRTIO_VSOCKETS=m
CONFIG_VIRTIO_VSOCKETS_COMMON=m
CONFIG_BT_VIRTIO=m
CONFIG_NET_9P_VIRTIO=m
CONFIG_VIRTIO_BLK=y
CONFIG_SCSI_VIRTIO=y
CONFIG_VIRTIO_NET=y
CONFIG_CAIF_VIRTIO=m
CONFIG_VIRTIO_CONSOLE=y
CONFIG_HW_RANDOM_VIRTIO=m
CONFIG_I2C_VIRTIO=m
CONFIG_GPIO_VIRTIO=m
CONFIG_DRM_VIRTIO_GPU=m
CONFIG_DRM_VIRTIO_GPU_KMS=y
CONFIG_SND_VIRTIO=m
CONFIG_VIRTIO_VFIO_PCI=m
CONFIG_VIRTIO_ANCHOR=y
CONFIG_VIRTIO=y
CONFIG_VIRTIO_PCI_LIB=y
CONFIG_VIRTIO_PCI_LIB_LEGACY=y
CONFIG_VIRTIO_MENU=y
CONFIG_VIRTIO_PCI=y
CONFIG_VIRTIO_PCI_ADMIN_LEGACY=y
CONFIG_VIRTIO_PCI_LEGACY=y
CONFIG_VIRTIO_VDPA=m
CONFIG_VIRTIO_PMEM=m
CONFIG_VIRTIO_BALLOON=y
CONFIG_VIRTIO_MEM=m
CONFIG_VIRTIO_INPUT=m
CONFIG_VIRTIO_MMIO=y
CONFIG_VIRTIO_MMIO_CMDLINE_DEVICES=y
CONFIG_VIRTIO_DMA_SHARED_BUFFER=m
CONFIG_XEN_VIRTIO=y
# CONFIG_XEN_VIRTIO_FORCE_GRANT is not set
CONFIG_VIRTIO_IOMMU=y
CONFIG_RPMSG_VIRTIO=m
CONFIG_VIRTIO_FS=m
CONFIG_CRYPTO_DEV_VIRTIO=m
```

**Comment by giuliazanchi** [comment]
Hi @vevayos , looks like you're getting into the depths of Multipass and its drivers! The output you're getting is correct, the driver used in your installation is QEMU. Check out this page to read more on the supported & default drivers, depending on the underlying system: https://multipass.run/docs/driver 

Take the time you need to see if this is something you feel comfortable contributing to, we can reassess next week or when you're ready :)

**Comment by vevayos** [comment]
Oh!!!  Thank you for the tip, @giuliazanchi!  I was confused whether I had QEMU or LXD.  I haven't used QEMU in years, and back in the day, I don't think QEMU responded with ICMP, so you couldn't ping the host from within the VM.  But I see that I can do that in my Multipass instance.

**Comment by vevayos** [comment]
Yikes, @giuliazanchi, someone was just mentioning sinking time on issues...  Well, this issue was still bugging me, and so I googled around...

I found this discussion related to what was confusing me...
[https://discuss.linuxcontainers.org/t/what-is-the-relationship-between-lxd-and-multipass/7758/2](https://discuss.linuxcontainers.org/t/what-is-the-relationship-between-lxd-and-multipass/7758/2)

This cleared up my issue with the main product page, [https://multipass.run/](https://multipass.run/), saying that Multipass uses an LXD hypervisor on Linux.

So on my system, the VMs are QEMU/KVM.  And so I'm guessing that experimental [libvirt](https://libvirt.org/) driver is probably being developed towards the goal that, "Upcoming Multipass will be using LXD VMs as backend on Linux."

Sorry to take up your time!  Thanks for your help!

**Comment by giuliazanchi** [comment]
Thanks for looking into this topic, @vevayos ! Take all the time you need, this is a complex and broad topic and not a quick fix, so it is not expected that you come up with a new, comprehensive section on Linux troubleshooting overnight :) Any input will be greatly appreciated!

**Comment by giuliazanchi** [comment]
Hi @vevayos , I've been doing some work to expand the Multipass Troubleshooting topics recently, so I thought I'd check in with you and see if you've had time to keep looking into this issue :) Let me know if you need help! You can also find me in the [Documentation room on Matrix](https://matrix.to/#/#documentation:ubuntu.com) if you want to chat.

**Comment by giuliazanchi** [comment]
HI @vevayos , are you still interested in contributing to this issue? If your availability has changed, no worries at all! Just let me know so I can free it up to other contributors :) Thanks!
